### PR TITLE
Ddp 5094 auth

### PR DESCRIPTION
**Pull request by charlottedye**
## Context

See DDP-5094.  This branch makes it possible for users to create accounts, log in and out of them, and change their passwords.  It also improves the UI for the authentication-related pages to make it clearer to the users that this is for the RGP study.  In addition, this changes the email for password change requests to look more like the email they may recognize from Gen2.  This also makes it possible to later add Spanish to the authentication pages and the change password email so that Spanish-speaking users will be able to use the site more easily.  In addition, this sets up the necessary configuration files to use the Auth0 Deploy CLI tool, which minimizes errors.

This is similar enough to existing studies that this should not affect performance.  It is possible that some of the changes to the RGP Auth0 pages will slow down loading of those pages themselves (particularly because of the need for additional JavaScript), but not any more than existing Auth0 pages.  

## Checklist

- [x] I have labeled the type of changes involved using the `C-*` labels.
- [x] I have assessed potential risks and labeled using the `R-*` labels.
- [x] I have considered error handling and alerts, and added `L-*` labels as needed.
- [x] I have considered security and privacy, and added `I-*` labels as needed
- [x] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.
- [x] If applicable, I have discussed the analytics needs at both a platform and study level with Product and instrumented code accordingly.
- [x] If applicable, my UI/UX changes have passed muster with Product/Design via an over-the-shoulder review, screenshots, etc.

_If unsure or need help with any of the above items, add the `help wanted` label. For items that starts with `If applicable`, if it is not applicable, check it off and add `n/a` in front._

## FUD Score

_Overall, how are you feeling about these changes?_

- [x] :relaxed: All good, business as usual!
- [ ] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [x] (Technically this requires the study to be deployed to dev, which is the next ticket) They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [ ] I have written automated positive tests
- [ ] I have written automated negative tests
- [x] I have written zero automated tests but have poked around locally to verify proper functionality
- [x] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [x] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

File: study-builder/config/manifest-rgp.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+render "application.conf.ctmpl"
+render "local.conf.ctmpl"
+render "vars.conf.ctmpl"
```
Comment (yufengwng) [inline study-builder/config/manifest-rgp.rb]: This is the same set of files we render in `manifest.rb`. Do we need this new file?

File: study-builder/tenants/rgp/emails/change_password.html [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+{% assign p5 = "The Rare Genomes Project Team" %}
+{% assign p6 = "If you did not make this request, please contact us." %}
+{% endif %}
+
+<center>
+    <table style="background: #ffffff; width: 600px; margin-top: 50px; padding: 0; -webkit-text-size-adjust: 100%;-ms-text-size-adjust: 100%; mso-table-lspace: 0pt; mso-table-rspace: 0pt; font-family: Roboto, sans-serif; border-collapse: collapse !important;"
+           align="center" border="0" cellpadding="0" cellspacing="0" width="100%">
+        <tr style="width: 600px;">
+            <td align="left" style="margin: 0; padding: 40px 0 40px 50px; font-family: Roboto, sans-serif; width: 175px;">
+                <img src="https://image.ibb.co/hSFqi5/logo.jpg" alt="Rare Genomes Project logo" style="-ms-interpolation-mode: bicubic;border: 0;height: auto; outline: none;text-decoration: none;">
```
Comment (yufengwng) [inline study-builder/tenants/rgp/emails/change_password.html]: Who's hosting this logo right now? Will it be better for us to host it so we ensure this URL won't be broken?

File: study-builder/tenants/rgp/emails/change_password.html [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+                <p style="padding: 25px 0 20px 0; text-align: center;">
+                    <a href="{{ url }}" style="background-color: #f55714 !important; color: #ffffff !important; font-family: 'Roboto', sans-serif; border-radius: 23px; font-weight: 500; font-size: 18px; padding: 13px 25px 12px 25px; cursor: pointer; text-align: center; text-decoration: none;">{{ p3 }}</a>
+                </p>
+
+                <p style="font-family: 'Roboto', sans-serif; text-align: left; font-weight: 300; font-size: 16px;">{{ p4 }}</p>
+                <p style="font-family: 'Roboto', sans-serif; text-align: left; font-weight: 500; padding-top: 10px; color: #0C457D; font-size: 16px;">{{ p5 }}</p>
+
+                <hr style="border: 2px solid #EAEEF3; border-bottom: 0; margin: 20px 0;">
+                <p style="font-family: 'Roboto', sans-serif; text-align: center; color: #A9B3BC;-webkit-text-size-adjust: 100%;-ms-text-size-adjust: 100%;font-size: 13px;">
+                   {{$p6}}
```
Comment (yufengwng) [inline study-builder/tenants/rgp/emails/change_password.html]: I think this is the wrong syntax?

File: study-builder/tenants/rgp/pages/login.html [hunk 4]
```diff
@@ -17,63 +18,337 @@
...
+        localStorage.setItem(inviteKey,config.extraParams.invitation_id);
+        console.log('Saving invitation ' + config.extraParams.invitation_id);
+    } else {
+        var invitation = localStorage.getItem(inviteKey);
+        if (invitation) {
+            // unsafe, but better than losing this value during breached password shield
+            config.internalOptions.invitation_id = invitation;
+            console.log('Using stored invitation ' + invitation + ' from local storage');
+        }
     }
```
Comment (yufengwng) [inline study-builder/tenants/rgp/pages/login.html]: I don't think this invitation_id business is applicable to RGP. Better to remove this so we don't run into potential issues.

File: study-builder/tenants/rgp/pages/login.html [hunk 5]
```diff
@@ -17,63 +18,337 @@
...
+
+    var mode = config.extraParams.mode;
+    var languageCode = config.extraParams.language;
+    var study_guid = config.extraParams.study_guid;
+    var studyColor = '#e8702a';
+    var studyHelpEmail = 'info@prionregistry.org';
```
Comment (yufengwng) [inline study-builder/tenants/rgp/pages/login.html]: You might want to take advantage of Auth0 CLI rendering and use `##STUDY_EMAIL##` here.

File: study-builder/tenants/rgp/pages/login.html [hunk 6]
```diff
@@ -17,63 +18,337 @@
 <script src="https://cdn.auth0.com/js/es5-shim.min.js"></script>
 <![endif]-->
 
-<script src="https://cdn.auth0.com/js/lock/11.27/lock.min.js"></script>
+<script src="https://cdn.auth0.com/js/lock/11.5.2/lock.min.js"></script>
```
Comment (yufengwng) [inline study-builder/tenants/rgp/pages/login.html]: Curious why we downgrade the version of auth0 lock library?

**Review by yufengwng — DISMISSED** [review]
Looks good. Just a few small comments.

File: study-builder/config/manifest-rgp.rb (hunk 1 above)
Comment (charlottedye) [reply study-builder/config/manifest-rgp.rb]: Good point.  I was thinking something wasn't working when I didn't have that file but I think that ended up being a different issue that I solved

File: study-builder/tenants/rgp/emails/change_password.html (hunk 2 above)
Comment (charlottedye) [reply study-builder/tenants/rgp/emails/change_password.html]: Thanks for catching this!

File: study-builder/tenants/rgp/emails/change_password.html (hunk 3 above)
Comment (charlottedye) [reply study-builder/tenants/rgp/emails/change_password.html]: Not sure how I missed this one!  I'm going to take a closer look when I test this time and compare side-by-side...

File: study-builder/tenants/rgp/pages/login.html (hunk 4 above)
Comment (charlottedye) [reply study-builder/tenants/rgp/pages/login.html]: Good point.  I went back and removed everything that isn't being used

File: study-builder/tenants/rgp/pages/login.html (hunk 5 above)
Comment (charlottedye) [reply study-builder/tenants/rgp/pages/login.html]: This is another thing I don't see being used

File: study-builder/tenants/rgp/pages/login.html (hunk 6 above)
Comment (charlottedye) [reply study-builder/tenants/rgp/pages/login.html]: I think I just copied one of the existing Auth0 login pages to give myself a head start and forgot to check the version.  I'll update the version and retest.

**Review by yufengwng — APPROVED** [review]

**Pull request merged**